### ISSUE TITLE
failure at wrong place(E1101 no-member: "Instance of 'list' has no 'update' member")

### ISSUE THREAD

**Issue by p00j4** [PyCQA/pylint#1355]
encountered in a big file and after debugging finally, got the reproducible pattern: 

```python
from mongoengine import *

class Query_Tester(Document):
    """
        Model
    """
    name = StringField()
    parameters = DictField()

    def method1(self):
        qt = Query_Tester()
        qt.parameters.update({"mfaset": "hello"})

    def method2(self):
        qt = Query_Tester()
        qt.parameters = ["mfaset", "hello"]  #no matter what order method1 & method2 are, it always gets set to list
```
### Current behavior
<img width="456" alt="screen shot 2017-03-01 at 7 09 33 pm" src="https://cloud.githubusercontent.com/assets/6470509/23462144/aa781cc2-feb2-11e6-969e-3c1150a2b6e7.png">

### Expected behavior
should give preference to the scope 
qt.parameters's scope should be in respective function 
 (or at-least to the sequence)

Even trying this leads to the same error  (should have ideally fail for reverse saying, its dict, can't assign like list)
```python
if __name__ == '__main__':
    qt = Query_Tester()
    qt.parameters.update({"gesture": "hello"})
    qt.parameters = ["hi", "hello"]
```
### pylint --version : 1.6.5

Yes practically, pylint does the job telling like both the places should be same type. but my worry is why does it point at wrong line. 
why i need: there are some places in my code-base which is like legacy now, if I get wrong line number pointer, gives "stars in daylight" moment, because i could clearly see in my scope, everywhere it was dict then why is it considering it a list. 
 I'm trying to enable lint check in ci, which makes the build failed and so the exercise of dig into is enforced. 
Any pointers I can dig more & work around?

**Comment by PCManticore** [comment]
Yes, no matter the order `pylint` is still going to emit `no-member` as it cannot infer the flow of your program, as such, it infers all the potential paths and all the potential values that that particular value can take. Unfortunately fixing this behaviour will require changing how pylint works conceptually, which is to be said, is not going to happen too soon.
